src/Platform/Microsoft.Testing.Platform/Hosts/TestHostBuilder.cs

```diff
@@ -283,10 +286,76 @@ internal class TestHostBuilder(IFileSystem fileSystem, IRuntimeFeature runtimeFe
             return toolsTestHost;
         }
 
+        NamedPipeClient? namedPipeClient = null;
```

Comment:
Can you refactor all this logic in a method like `ConnectToTestHostProcessMonitorIfAvailableAsync`

something like `NamedPipeClient? ConnectToDotnetTestPipeIfAvailableAsync(...);`

```
// If we are in server mode and we the pipe name is provided
// then, we need to connect to the pipe server.
NamedPipeClient? dotnetTestPipeClient = ConnectToDotnetTestPipeIfAvailableAsync(...);
```